Fix sequential MKL case

ilia-cher opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #22114 Fix in ivalue::Future
* **#22062 Fix sequential MKL case**

Summary:
Make sure we don't reset number of OMP threads to 1
in case of sequential MKL library

Test Plan:
USE_CUDA=0 PARALLEL_BACKEND=OPENMP BLAS=MKL USE_MKLDNN=1 MKL_SEQ=1
MKLDNN_THREADING=SEQ BUILD_BINARY=1 python setup.py develop --cmake

./build/bin/parallel_info

Differential Revision: [D15938079](https://our.internmc.facebook.com/intern/diff/D15938079)

Commit:
Fix sequential MKL case

Summary:
Make sure we don't reset number of OMP threads to 1
in case of sequential MKL library

Test Plan:
USE_CUDA=0 PARALLEL_BACKEND=OPENMP BLAS=MKL USE_MKLDNN=1 MKL_SEQ=1
MKLDNN_THREADING=SEQ BUILD_BINARY=1 python setup.py develop --cmake

./build/bin/parallel_info

cpuhrsch commented on CMakeLists.txt:
```diff
@@ -139,6 +139,8 @@ cmake_dependent_option(
     "USE_GLOO" OFF)
 option(USE_TBB "Use TBB" OFF)
 option(USE_EIGEN_THREADPOOL, "Use Eigen thread pool" OFF)
+option(USE_MKL_SEQ "Use sequential MKL" OFF)
```
Instead of this maybe make it flag you pass a string to? Makes it easier to keep an overview too and warn on a wrong setting.

Commit:
Update on "Fix sequential MKL case"

Fix sequential MKL case

Summary:
Make sure we don't reset number of OMP threads to 1
in case of sequential MKL library

Test Plan:
USE_CUDA=0 PARALLEL_BACKEND=OPENMP BLAS=MKL USE_MKLDNN=1 MKL_SEQ=1
MKLDNN_THREADING=SEQ BUILD_BINARY=1 python setup.py develop --cmake

./build/bin/parallel_info

gh-metadata: pytorch pytorch 22062 gh/ilia-cher/33/head

Commit:
Update on "Fix sequential MKL case"

Fix sequential MKL case

Summary:
Make sure we don't reset number of OMP threads to 1
in case of sequential MKL library

Test Plan:
USE_CUDA=0 PARALLEL_BACKEND=OPENMP BLAS=MKL USE_MKLDNN=1 MKL_SEQ=1
MKLDNN_THREADING=SEQ BUILD_BINARY=1 python setup.py develop --cmake

./build/bin/parallel_info

gh-metadata: pytorch pytorch 22062 gh/ilia-cher/33/head

Commit:
Update on "Fix sequential MKL case"

Fix sequential MKL case

Summary:
Make sure we don't reset number of OMP threads to 1
in case of sequential MKL library

Test Plan:
USE_CUDA=0 PARALLEL_BACKEND=OPENMP BLAS=MKL USE_MKLDNN=1 MKL_SEQ=1
MKLDNN_THREADING=SEQ BUILD_BINARY=1 python setup.py develop --cmake

./build/bin/parallel_info

gh-metadata: pytorch pytorch 22062 gh/ilia-cher/33/head

Commit:
Update on "Fix sequential MKL case"

Fix sequential MKL case

Summary:
Make sure we don't reset number of OMP threads to 1
in case of sequential MKL library

Test Plan:
USE_CUDA=0 PARALLEL_BACKEND=OPENMP BLAS=MKL USE_MKLDNN=1 MKL_SEQ=1
MKLDNN_THREADING=SEQ BUILD_BINARY=1 python setup.py develop --cmake

./build/bin/parallel_info

gh-metadata: pytorch pytorch 22062 gh/ilia-cher/33/head

Commit:
Update on "Fix sequential MKL case"

Fix sequential MKL case

Summary:
Make sure we don't reset number of OMP threads to 1
in case of sequential MKL library

Test Plan:
USE_CUDA=0 PARALLEL_BACKEND=OPENMP BLAS=MKL USE_MKLDNN=1 MKL_SEQ=1
MKLDNN_THREADING=SEQ BUILD_BINARY=1 python setup.py develop --cmake

./build/bin/parallel_info

gh-metadata: pytorch pytorch 22062 gh/ilia-cher/33/head

Commit:
Update on "Fix sequential MKL case"

Fix sequential MKL case

Summary:
Make sure we don't reset number of OMP threads to 1
in case of sequential MKL library

Test Plan:
USE_CUDA=0 PARALLEL_BACKEND=OPENMP BLAS=MKL USE_MKLDNN=1 MKL_SEQ=1
MKLDNN_THREADING=SEQ BUILD_BINARY=1 python setup.py develop --cmake

./build/bin/parallel_info

gh-metadata: pytorch pytorch 22062 gh/ilia-cher/33/head

Commit:
Update on "Fix sequential MKL case"

Fix sequential MKL case

Summary:
Make sure we don't reset number of OMP threads to 1
in case of sequential MKL library

Test Plan:
USE_CUDA=0 PARALLEL_BACKEND=OPENMP BLAS=MKL USE_MKLDNN=1 MKL_SEQ=1
MKLDNN_THREADING=SEQ BUILD_BINARY=1 python setup.py develop --cmake

./build/bin/parallel_info

gh-metadata: pytorch pytorch 22062 gh/ilia-cher/33/head

Commit:
Update on "Fix sequential MKL case"

Fix sequential MKL case

Summary:
Make sure we don't reset number of OMP threads to 1
in case of sequential MKL library

Test Plan:
USE_CUDA=0 PARALLEL_BACKEND=OPENMP BLAS=MKL USE_MKLDNN=1 MKL_SEQ=1
MKLDNN_THREADING=SEQ BUILD_BINARY=1 python setup.py develop --cmake

./build/bin/parallel_info

gh-metadata: pytorch pytorch 22062 gh/ilia-cher/33/head

Commit:
Update on "Fix sequential MKL case"

Fix sequential MKL case

Summary:
Make sure we don't reset number of OMP threads to 1
in case of sequential MKL library

Test Plan:
USE_CUDA=0 PARALLEL_BACKEND=OPENMP BLAS=MKL USE_MKLDNN=1 MKL_SEQ=1
MKLDNN_THREADING=SEQ BUILD_BINARY=1 python setup.py develop --cmake

./build/bin/parallel_info

gh-metadata: pytorch pytorch 22062 gh/ilia-cher/33/head

Commit:
Update on "Fix sequential MKL case"

Fix sequential MKL case

Summary:
Make sure we don't reset number of OMP threads to 1
in case of sequential MKL library

Test Plan:
USE_CUDA=0 PARALLEL_BACKEND=OPENMP BLAS=MKL USE_MKLDNN=1 MKL_SEQ=1
MKLDNN_THREADING=SEQ BUILD_BINARY=1 python setup.py develop --cmake

./build/bin/parallel_info

gh-metadata: pytorch pytorch 22062 gh/ilia-cher/33/head

Commit:
Update on "Fix sequential MKL case"

Fix sequential MKL case

Summary:
Make sure we don't reset number of OMP threads to 1
in case of sequential MKL library

Test Plan:
USE_CUDA=0 PARALLEL_BACKEND=OPENMP BLAS=MKL USE_MKLDNN=1 MKL_SEQ=1
MKLDNN_THREADING=SEQ BUILD_BINARY=1 python setup.py develop --cmake

./build/bin/parallel_info

gh-metadata: pytorch pytorch 22062 gh/ilia-cher/33/head

Commit:
Update on "Fix sequential MKL case"

Fix sequential MKL case

Summary:
Make sure we don't reset number of OMP threads to 1
in case of sequential MKL library

Test Plan:
USE_CUDA=0 PARALLEL_BACKEND=OPENMP BLAS=MKL USE_MKLDNN=1 MKL_SEQ=1
MKLDNN_THREADING=SEQ BUILD_BINARY=1 python setup.py develop --cmake

./build/bin/parallel_info

gh-metadata: pytorch pytorch 22062 gh/ilia-cher/33/head

Commit:
Update on "Fix sequential MKL case"

Fix sequential MKL case

Summary:
Make sure we don't reset number of OMP threads to 1
in case of sequential MKL library

Test Plan:
USE_CUDA=0 PARALLEL_BACKEND=OPENMP BLAS=MKL USE_MKLDNN=1 MKL_SEQ=1
MKLDNN_THREADING=SEQ BUILD_BINARY=1 python setup.py develop --cmake

./build/bin/parallel_info

gh-metadata: pytorch pytorch 22062 gh/ilia-cher/33/head

cpuhrsch reviewed (approved):
cool
